### PR TITLE
Time Series: adjust card width (UI)

### PR DESCRIPTION
**Pull request by japie1235813**
* Motivation for features / changes
Some tag name on cards might be too long and be truncated. We want to provide users a setting to adjust the card width. This pr only includes the part of metrics/ and the card with would be set based on the state value.

* Technical description of changes
Only update the max width of the card in css. The range of the max width is 30-100vw.

### CODE REVIEW

File: tensorboard/webapp/metrics/store/metrics_selectors.ts [hunk 1]
```diff
@@ -280,6 +280,11 @@ export const getMetricsSettingOverrides = createSelector(
   }
 );
 
+export const getMetricCardMaxWidth = createSelector(
```
Comment (yatbear) [inline tensorboard/webapp/metrics/store/metrics_selectors.ts:283]: Should this be getMetric[s]CardMaxWidth to be consistent with the other ones?

Comment (japie1235813) [reply tensorboard/webapp/metrics/store/metrics_selectors.ts]: good catch. done

File: tensorboard/webapp/metrics/store/metrics_types.ts [hunk 2]
```diff
@@ -221,6 +222,7 @@ export interface State {
 }
 
 export const METRICS_SETTINGS_DEFAULT: MetricsSettings = {
+  cardMaxWidthInVW: 30,
```
[the remark this answers is not in the document]
Comment (japie1235813) [reply tensorboard/webapp/metrics/store/metrics_types.ts]: Thanks for catching this. It should be null by default

File: tensorboard/webapp/metrics/views/main_view/card_grid_component.ts [hunk 3]
```diff
@@ -49,6 +55,16 @@ export class CardGridComponent {
     @Optional() private readonly cdkScrollable: CdkScrollable | null
   ) {}
 
+  ngOnInit() {
+    if (
+      this.cardMaxWidthInVW &&
+      this.cardMaxWidthInVW >= MIN_CARD_MAX_WIDTH_IN_VW &&
```
[the remark this answers is not in the document]
Comment (japie1235813) [reply tensorboard/webapp/metrics/views/main_view/card_grid_component.ts:61]: Short answer is no.  The range of the slider should be 30 to 100 (unit vw). And we will not show the numbers to preserve interpretation (instead of explicitly explaining what the numbers are)

File: tensorboard/webapp/metrics/views/main_view/main_view_test.ts [hunk 4]
```diff
@@ -894,6 +895,75 @@ describe('metrics main view', () => {
...
+        fixture.detectChanges();
+
+        expect(
+          fixture.debugElement.query(By.css('.card-grid')).styles[
+            'grid-template-columns'
+          ]
+        ).toBe('repeat(auto-fill, minmax(335px, 50vw))');
+      });
+
+      it('does not set the max width with invalid width value', () => {
```
Comment (bmd3k) [inline tensorboard/webapp/metrics/views/main_view/main_view_test.ts:935]: Just for completeness, and given that it will be the default behavior, would you mind adding a test where getEnabledCardWidthSetting is true and getMetricsCardWidth is null?

**Review by bmd3k — APPROVED** [review]
